In [1]:
import numpy as np
import pandas as pd

In [2]:
train_data = pd.read_csv("/content/train.csv")
train_data

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [3]:
df_test = pd.read_csv("/content/test.csv")
df_test

PassengerId  Pclass                                          Name  \
0            892       3                              Kelly, Mr. James   
1            893       3              Wilkes, Mrs. James (Ellen Needs)   
2            894       2                     Myles, Mr. Thomas Francis   
3            895       3                              Wirz, Mr. Albert   
4            896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
..           ...     ...                                           ...   
413         1305       3                            Spector, Mr. Woolf   
414         1306       1                  Oliva y Ocana, Dona. Fermina   
415         1307       3                  Saether, Mr. Simon Sivertsen   
416         1308       3                           Ware, Mr. Frederick   
417         1309       3                      Peter, Master. Michael J   

        Sex   Age  SibSp  Parch              Ticket      Fare Cabin Embarked  
0      male  34.5      0      0              330911    7.8292   NaN        Q  
1    female  47.0      1      0              363272    7.0000   NaN        S  
2      male  62.0      0      0              240276    9.6875   NaN        Q  
3      male  27.0      0      0              315154    8.6625   NaN        S  
4    female  22.0      1      1             3101298   12.2875   NaN        S  
..      ...   ...    ...    ...                 ...       ...   ...      ...  
413    male   NaN      0      0           A.5. 3236    8.0500   NaN        S  
414  female  39.0      0      0            PC 17758  108.9000  C105        C  
415    male  38.5      0      0  SOTON/O.Q. 3101262    7.2500   NaN        S  
416    male   NaN      0      0              359309    8.0500   NaN        S  
417    male   NaN      1      1                2668   22.3583   NaN        C  

[418 rows x 11 columns]

droping some features those are less important as per basic intuition.
making model simple but effective

In [4]:
df_train=train_data.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'],axis='columns')#inplace
df_train

Survived  Pclass     Sex   Age     Fare
0           0       3    male  22.0   7.2500
1           1       1  female  38.0  71.2833
2           1       3  female  26.0   7.9250
3           1       1  female  35.0  53.1000
4           0       3    male  35.0   8.0500
..        ...     ...     ...   ...      ...
886         0       2    male  27.0  13.0000
887         1       1  female  19.0  30.0000
888         0       3  female   NaN  23.4500
889         1       1    male  26.0  30.0000
890         0       3    male  32.0   7.7500

[891 rows x 5 columns]

Detrmine inputs and output

In [5]:
inputs = df_train.drop('Survived',axis='columns')
inputs

Pclass     Sex   Age     Fare
0         3    male  22.0   7.2500
1         1  female  38.0  71.2833
2         3  female  26.0   7.9250
3         1  female  35.0  53.1000
4         3    male  35.0   8.0500
..      ...     ...   ...      ...
886       2    male  27.0  13.0000
887       1  female  19.0  30.0000
888       3  female   NaN  23.4500
889       1    male  26.0  30.0000
890       3    male  32.0   7.7500

[891 rows x 4 columns]

In [6]:
target = df_train.Survived
target

0      0
1      1
2      1
3      1
4      0
      ..
886    0
887    1
888    0
889    1
890    0
Name: Survived, Length: 891, dtype: int64

Get dummies method pandas

In [7]:
dummies = pd.get_dummies(inputs.Sex  )
dummies.head()

female  male
0       0     1
1       1     0
2       1     0
3       1     0
4       0     1

In [8]:
inputs = pd.concat([inputs,dummies],axis='columns')
inputs

Pclass     Sex   Age     Fare  female  male
0         3    male  22.0   7.2500       0     1
1         1  female  38.0  71.2833       1     0
2         3  female  26.0   7.9250       1     0
3         1  female  35.0  53.1000       1     0
4         3    male  35.0   8.0500       0     1
..      ...     ...   ...      ...     ...   ...
886       2    male  27.0  13.0000       0     1
887       1  female  19.0  30.0000       1     0
888       3  female   NaN  23.4500       1     0
889       1    male  26.0  30.0000       0     1
890       3    male  32.0   7.7500       0     1

[891 rows x 6 columns]

I am dropping male column as well because of dummy variable trap theory. One column is enough to repressent male vs female



In [9]:
inputs.drop(['Sex','male'],axis='columns',inplace=True)
inputs

Pclass   Age     Fare  female
0         3  22.0   7.2500       0
1         1  38.0  71.2833       1
2         3  26.0   7.9250       1
3         1  35.0  53.1000       1
4         3  35.0   8.0500       0
..      ...   ...      ...     ...
886       2  27.0  13.0000       0
887       1  19.0  30.0000       1
888       3   NaN  23.4500       1
889       1  26.0  30.0000       0
890       3  32.0   7.7500       0

[891 rows x 4 columns]

**checking missing values**

In [10]:
inputs.isnull().sum()

Pclass      0
Age       177
Fare        0
female      0
dtype: int64

**here age has 177 missing values. Filling missing values with meadian value of age column.**

In [11]:
inputs['Age'].median()

28.0

In [12]:
inputs.Age = inputs.Age.fillna(inputs.Age.median())
inputs

Pclass   Age     Fare  female
0         3  22.0   7.2500       0
1         1  38.0  71.2833       1
2         3  26.0   7.9250       1
3         1  35.0  53.1000       1
4         3  35.0   8.0500       0
..      ...   ...      ...     ...
886       2  27.0  13.0000       0
887       1  19.0  30.0000       1
888       3  28.0  23.4500       1
889       1  26.0  30.0000       0
890       3  32.0   7.7500       0

[891 rows x 4 columns]

In [13]:
inputs.isnull().sum()

Pclass    0
Age       0
Fare      0
female    0
dtype: int64

**importing and building model**

In [ ]:
from sklearn.naive_bayes import GaussianNB
model = GaussianNB()

In [ ]:
model.fit(inputs, target)

GaussianNB()

In [ ]:
test_data1 = pd.read_csv("/content/test.csv")
test_data1.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

In [ ]:
test_data1.shape

(418, 11)

**making X_test and y_test**

In [ ]:
test_data = test_data1.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'],axis='columns')
test_data.head()

Pclass     Sex   Age     Fare
0       3    male  34.5   7.8292
1       3  female  47.0   7.0000
2       2    male  62.0   9.6875
3       3    male  27.0   8.6625
4       3  female  22.0  12.2875

In [ ]:
dummies_test = pd.get_dummies(test_data.Sex  )
dummies_test.head()

female  male
0       0     1
1       1     0
2       0     1
3       0     1
4       1     0

In [ ]:
X_test = pd.concat([test_data, dummies_test],axis='columns')
X_test.drop(['Sex','male'],axis='columns',inplace=True)
X_test.head()

Pclass   Age     Fare  female
0       3  34.5   7.8292       0
1       3  47.0   7.0000       1
2       2  62.0   9.6875       0
3       3  27.0   8.6625       0
4       3  22.0  12.2875       1

In [ ]:
X_test.isnull().sum()

Pclass     0
Age       86
Fare       1
female     0
dtype: int64

In [ ]:
X_test.Fare.mode()

0    7.75
dtype: float64

In [ ]:
X_test.Age = X_test.Age.fillna(X_test.Age.median())
X_test.Fare = X_test.Fare.fillna(X_test.Fare.mean())
X_test.head()

Pclass   Age     Fare  female
0       3  34.5   7.8292       0
1       3  47.0   7.0000       1
2       2  62.0   9.6875       0
3       3  27.0   8.6625       0
4       3  22.0  12.2875       1

In [ ]:
X_test[X_test['Fare'].isnull()]

Empty DataFrame
Columns: [Pclass, Age, Fare, female]
Index: []

In [ ]:
y_predicted = model.predict(X_test)
y_predicted

array([0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0,
       1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1,
       1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1,
       1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1,
       1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0,
       1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1,
       1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0,
       0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,

In [ ]:
test_data1['PassengerId']

0       892
1       893
2       894
3       895
4       896
       ... 
413    1305
414    1306
415    1307
416    1308
417    1309
Name: PassengerId, Length: 418, dtype: int64

In [ ]:
submission_dataframe = pd.DataFrame(data = test_data1['PassengerId'], columns = ['PassengerId'])

In [ ]:
submission_dataframe['Survived'] = y_predicted
submission_dataframe

PassengerId  Survived
0            892         0
1            893         1
2            894         0
3            895         0
4            896         1
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[418 rows x 2 columns]

In [ ]:
submission_dataframe.to_csv('submission.csv', index = False)
